### PR TITLE
Cameras now have 5x5 and 7x7 by default.

### PR DESCRIPTION
**Pull request by ghost**
Adds 5x5 and 7x7 to the normal camera for our hard working reporters, stopping at nothing to bring YOU the latest in space news.

Details:
- Basic/sepia cameras now switches between 1x1, 3x3, 5x5, and 7x7 (alt-click or verb).
- 5x5 and 7x7 pictures are larger than the normal-sized pictures. Newscaster images are static size but made to be slightly larger. Perfect for a shot of the singularity eating the station!
- Flavor text adjusted accordingly.
- Removes 5x5 camera (previously only found in Spessmart and Assistant's Lair).
- Cleaned up camera code a bit.

Possible Issues:
- (Fixed, see comments) _Lag. The 7x7 works on my machine good, but apparently 5x5 was removed from the server at an earlier point due to lag. If anyone could test that out, I'd appreciate it. If it causes lag, I'll take a look at whatever abomination BYOND has for a .png encoding library and see what I can do._

Other notes:
- Doesn't touch AI/cyborg cameras. 

:cl: 
* tweak: All non-silicon cameras have 5x5 and 7x7 modes now (alt click)
* rscdel: Deleted existing 5x5 cameras

### CODE REVIEW

**Comment by PaulRitter** [comment]
YES, altough maybe not if this actually lags

also add :cl:

**Comment by ShiftyRail** [comment]
I was under the belief that this explicitly wasn't an option due to how costly this could be

**Comment by Exxion** [comment]
Yeah they were left out on purpose because they're super fucking laggy
It depends heavily on what's in the picture but if it's chaotic enough the big camera can easily hang the server for half a second

EDIT: Oh, as for looking into the PNG encoding library: It doesn't exist but that's the least of your worries
While converting the image to a PNG is significantly costly, the much worse part is creating the image in the first place thanks to how abysmally fucking slow BYOND's icon procs are

**Comment by ghost** [comment]
Alright, so what I did to fix potential lag issues was make 5x5 and 7x7 require time to develop. 
3x3 = 9 tiles
5x5 = 25 tiles
7x7 = 49 tiles
Now every 10 tiles it triggers a .1 second break. 3x3's lag is almost unnoticable, so this allows us to get the bigger pictures without a noticeable microstutter. I still haven't tested it with real server conditions (obviously) so my recommendation would be to test it out on bleeding-edge and then revert if it causes any issues. I don't expect it to though.

**Comment by Exxion** [comment]
OK, that's fair
It *is* possible to make the 3x3 lag noticeably but it's hard to imagine a case where it would happen by accident

File: code/modules/paperwork/photography.dm [hunk 1]
```diff
@@ -305,6 +315,9 @@
 	var/icon/res = get_base_photo_icon()
 
 	for(var/atom/A in plane_layer_sort(atoms))
+		iterator++
+		if(iterator % tickspersleep == 0)
+			sleep(1)
```
Comment (ShiftyRail) [inline code/modules/paperwork/photography.dm]: Idea for this : `CHECK_TICK`, which makes it sleep only if the CPU used by the server is currently at or above a certain threshold.

**Comment by ValkyrieSkies** [comment]
You can testmerge PRs to the Test Server via MoMMIv2 on the coder discord.

File: code/modules/paperwork/photography.dm (hunk 1 above)
Comment (ghost) [reply code/modules/paperwork/photography.dm]: Tried this out and seems to work basically the same but in theory should be more smooth, so I changed it.

File: code/modules/paperwork/photography.dm [hunk 2]
```diff
@@ -305,6 +313,8 @@
 	var/icon/res = get_base_photo_icon()
 
 	for(var/atom/A in plane_layer_sort(atoms))
+		iterator++
```
Comment (ShiftyRail) [inline code/modules/paperwork/photography.dm]: If you're using `CHECK_TICK`, then the iterator is no longer defined

Comment (ghost) [reply code/modules/paperwork/photography.dm]: Yeah, I didn't compile before submitting. Just fixed it, my bad

**Comment by Kurfursten** [comment]
Add 🆑

**Comment by ghost** [comment]
Added, just joined coding community so what exactly does that signify? Just that it's ready to merge?

**Comment by ShiftyRail** [comment]
:cl: on your original post ensures that your PR will automagically get a changelog in the game.
Like this :
:cl:
- rscadd: did stuff

**Comment by Kurfursten** [comment]
I've edited your OP so you can see how to do it in the future, basically I wanted you to add the part below the 🆑 emoji (emoji included). This is what appears in the changelog.

**Pull request merged**